hazelcast/src/main/java/com/hazelcast/topic/ITopic.java

```diff
@@ -63,6 +66,16 @@
      */
     void publish(@Nonnull E message);
 
+    /**
+     * Publishes the message to all subscribers of this topic.
+     *
+     * @param message the message to publish to all subscribers of this topic
+     * @return the CompletionStage to synchronize on completion.
+     * @throws TopicOverloadException if the consumer is too slow
```

Comment:
Do we actually throw it or do we complete the future with this exception?

Reply:
Ok, you are right. The future is completed with this exception `returnFuture.completeExceptionally(t);`
I removed the @throws.